include/sparrow/typed_array.hpp

```diff
@@ -0,0 +1,360 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <compare>
+
+#include "sparrow/algorithm.hpp"
+#include "sparrow/array_data.hpp"
+#include "sparrow/data_traits.hpp"
+#include "sparrow/data_type.hpp"
+
+#if !defined(COMPILING_WITH_APPLE_CLANG)
+#include <format>
+#endif
+
+namespace sparrow
+{
+    template <class T, class L = typename arrow_traits<T>::default_layout>
+        requires is_arrow_base_type<T>
+    class typed_array
```

Comment:
The documentation for this type is missing (I mean the overview of the type, comments before it's definition)